lighthouse-cli/bin.ts

```diff
@@ -272,82 +251,92 @@ function handleError(err: LighthouseError) {
-        return Printer.write(results, flags.output, outputPath).then(results => {
-          if (flags.output === Printer.OutputMode[Printer.OutputMode.html] ||
-              flags.output === Printer.OutputMode[Printer.OutputMode.domhtml]) {
-            if (flags.view) {
-              opn(outputPath, {wait: false});
-            } else {
-              log.log('CLI', 'Protip: Run lighthouse with `--view` to immediately open the HTML report in your browser');
-            }
+  const typeToExtension = (type: string) => type === 'domhtml' ? 'dom.html' : type;
+  return promise.then(_ => {
+    if (Array.isArray(flags.output)) {
+      return flags.output.reduce((innerPromise, outputType) => {
+        const outputPath = `${resolvedPath}.report.${typeToExtension(outputType)}`;
+        return innerPromise.then((_: Results) => Printer.write(results, outputType, outputPath));
+      }, Promise.resolve(results));
+    } else {
+      const outputPath =
+          flags.outputPath || `${resolvedPath}.report.${typeToExtension(flags.output)}`;
+      return Printer.write(results, flags.output, outputPath).then(results => {
+        if (flags.output === Printer.OutputMode[Printer.OutputMode.html] ||
+            flags.output === Printer.OutputMode[Printer.OutputMode.domhtml]) {
+          if (flags.view) {
+            opn(outputPath, {wait: false});
+          } else {
+            log.log(
+                'CLI',
+                'Protip: Run lighthouse with `--view` to immediately open the HTML report in your browser');
           }
+        }
 
-          return results;
-        });
-      }
-    });
+        return results;
+      });
+    }
+  });
 }
 
-export async function runLighthouse(url: string,
-                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any,
-                         outputPath: string, interactive: boolean, saveArtifacts: boolean, saveAssets: boolean
-                         chromeFlags: string, maxWaitForLoad: number, view: boolean},
-                       config: Object | null): Promise<{}|void> {
-
-  let chromeLauncher: ChromeLauncher | undefined = undefined;
-
-  try {
-    await initPort(flags)
-    const chromeLauncher = await getDebuggableChrome(flags)
-    const results = await lighthouse(url, flags, config);
-
-    const artifacts = results.artifacts;
-    delete results.artifacts;
-
-    await saveResults(results, artifacts!, flags);
-    if (flags.interactive) {
-      await performanceXServer.hostExperiment({url, flags, config}, results);
-    }
+export async function runLighthouse(
+  url: string,
+  flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any,
+    outputPath: string, interactive: boolean, saveArtifacts: boolean, saveAssets: boolean
+    chromeFlags: string, maxWaitForLoad: number, view: boolean},
+    config: Object | null): Promise<{}|void> {
+      let chromeLauncher: ChromeLauncher|undefined = undefined;
```

Comment:
It is due to the large object map for args, since it tried to indent 1 block level in. In my follow up cl that cleans up the flags interface this is totally 100% better.